zerver/views/message_flags.py

```diff
@@ -50,7 +51,10 @@ def update_message_flags(
 @has_request_variables
 def mark_all_as_read(request: HttpRequest, user_profile: UserProfile) -> HttpResponse:
     request_notes = RequestNotes.get_notes(request)
-    count = do_mark_all_as_read(user_profile)
+    try:
+        count = timeout(50, lambda: do_mark_all_as_read(user_profile))
+    except TimeoutExpired:
+        return json_response(msg="partial success", status=206)
```

Comment:
How about this? @alexmv  It seems kind of nice to distinguish this situation with a different status code and `206` seemed like a good suggestion :thinking: This is also eliminates the need for additional code in `unread_ops.js`, since `206` is still `success, so just added a comment there.

Reply:
Do we want to add a `code` to this class of response, and define it in `zerver/lib/exceptions.py`? It might be helpful to do so both as a pattern we can reuse and also to make it possible to change that "partial success" text.

Reply:
I like the idea of turning this into a JsonableError with `http_status_code = 206` and a `code`.